tests/unit/test_experiments/test_simulation_with_experiment.py

```diff
@@ -438,6 +438,111 @@ def test_save_at_cycles(self):
         # Summary variables are not None
         self.assertIsNotNone(sol.summary_variables["Capacity [A.h]"])
 
+    def test_experiment_solution_step_tag_filters(self):
+        # Test cycle_summary_variables works for different combinations of data and
+        # function OCPs
+        experiment = pybamm.Experiment(
+            [
+                (
+                    pybamm.step.string(
+                        "Discharge at 1C until 3.3V", tags=["tag1", "tag2"]
+                    )
+                ),
+                (
+                    pybamm.step.string("Charge at C/3 until 4.0V", tags=["tag3"]),
+                    pybamm.step.string("Hold at 4.0V until C/10", tags=["tag4"]),
+                ),
+            ]
+            * 5,
+        )
+        model = pybamm.lithium_ion.SPM()
+
+        # O'Kane 2022: pos = function, neg = data
+        param = pybamm.ParameterValues("OKane2022")
+        sim = pybamm.Simulation(model, experiment=experiment, parameter_values=param)
+        solution = sim.solve(solver=pybamm.CasadiSolver("fast with events"))
+
+        sub_solutions = solution.search_tags(["tag1"])
+        assert len(sub_solutions) == 5
+
+        sub_solutions = solution.search_tags(["tag3"])
+        assert len(sub_solutions) == 5
+
+        sub_solutions = solution.search_tags(["tag1", "tag3"])
+        assert len(sub_solutions) == 10
+
+    def test_experiment_solution_step_tags(self):
```

Comment:
I don't think this test adds any additional checks compared to the previous test?

Reply:
I think you are correct. I will add both to a single test.